libsolidity/formal/CHC.cpp

```diff
@@ -75,16 +77,42 @@ void CHC::analyze(SourceUnit const& _source)
 	m_context.setAssertionAccumulation(false);
 	m_variableUsage.setFunctionInlining(false);
 
+	resetSourceAnalysis();
+
 	auto boolSort = make_shared<smt::Sort>(smt::Kind::Bool);
 	auto genesisSort = make_shared<smt::FunctionSort>(
 		vector<smt::SortPointer>(),
 		boolSort
 	);
 	m_genesisPredicate = createSymbolicBlock(genesisSort, "genesis");
-	auto genesis = (*m_genesisPredicate)({});
-	addRule(genesis, genesis.name);
+	addRule(genesis(), "genesis");
+
+	defineInterfacesAndSummaries(_source);
+	for (auto const& source: _source.referencedSourceUnits())
```

Comment:
Why would it be nondet?

Sounds good anyway.

Reply:
Would you say that because if the referenced source units visit, the toAnalyzeContracts member is redundant?

Reply:
The order of referenced source units is nondet, since it's a set of pointers. First I thought that might not be relevant, but ``defineInterfacesAndSummaries`` calls ``createSummaryBlock`` which calls ``uniquePrefix()`` to generate names, which is a simple counter. So you'll end up with non-deterministic names for the summaries and I'm pretty sure that way we loose all determinism-guarantees of the solvers.

Reply:
Right. I applied your suggestion